### PR TITLE
Add a few more swagger descriptions and examples.

### PR DESCRIPTION
**Pull request by fisx**
2024-10-31-134844_1600x900_scrot
2024-10-31-134825_1600x900_scrot
2024-10-31-134816_1600x900_scrot

## Checklist

 - [X] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [X] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Comment by mastaab** [comment]
can you maybe also add a description for "currency" while you are at it?

**Comment by netbe** [comment]
out of curiousity, where does the retention_information link go to ? @fisx

File: libs/wire-api/src/Wire/API/Asset.hs [hunk 1]
```diff
@@ -177,8 +177,13 @@ assetKeyToText = T.decodeUtf8 . toByteString'
...
   schema =
     assetKeyToText
-      .= parsedText "AssetKey" (runParser parser . T.encodeUtf8)
+      .= parsedTextWithDoc desc "AssetKey" (runParser parser . T.encodeUtf8)
       & doc' . S.schema . S.example ?~ toJSON ("3-1-47de4580-ae51-4650-acbb-d10c028cb0ac" :: Text)
+    where
+      desc =
+        "S3 asset key for the encrypted icon image with <a href=\"https://github.com/wireapp/\
+        \wire-server/blob/734588952d87e76208d1fa15800a5cfb76d541ee/libs/wire-api/src/Wire/API/\
+        \Asset.hs#L148-L164\">retention information</a>."
```
Comment (akshaymankar) [inline libs/wire-api/src/Wire/API/Asset.hs:186]: Its not always encrypted, there are assets which are plain like team icon. 
That link will likely get stale, also linking to haskell code is probably not very helpful anyway.

**Comment by fisx** [comment]
> out of curiousity, where does the retention_information link go to ? @fisx

the code that parses asset ids :-)

but as @akshaymankar says, it's probably better to take that out again.

**Comment by fisx** [comment]
[after.json](https://github.com/user-attachments/files/17600624/after.json)
[before.json](https://github.com/user-attachments/files/17600637/before.json)
[diff.json](https://github.com/user-attachments/files/17600638/diff.json)

**Pull request merged**